File: Server/Models/Employee.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Server.Models;

public class Employee : Person
{
    public string Position { get; set; }
    public int Department { get; set; }  // Department object
    public string ImageUrl { get; set; }
    public int Permission { get; set; }  // Permission object

    public Employee()
    {
        
    }

    public Employee(string name, int age, string address, int employeeId, string position, int department, string imageUrl, int permission)
        : base(employeeId, name, age, address)
    {
        Position = position;
        Department = department;
        ImageUrl = imageUrl;
        Permission = permission;
    }



    public override string ToString()
    {
        return $"{base.ToString()}, Employee ID: {Id}, Position: {Position}, {Department}, {Permission}";
    }
}

File: Client/Models/Employee.py
from models.person import Person
from models.department import Department
from models.permission import Permission

# Employee model class, extending Person with additional employee-specific attributes.

class Employee(Person):
    def __init__(self, position, department, imageUrl, permission, id, name, age, address):
        super().__init__(id, name, age, address)
        self.position = position
        self.department_id = department
        self.imageUrl = imageUrl
        self.permission_id = permission

    def to_dict(self):
        return super().to_dict() | {
            "Position": self.position,
            "Department": self.department_id,
            "ImageUrl": "Url",
            "Permission": self.permission_id
        }

